lib/generators/sufia/templates/catalog_controller.rb

```diff
@@ -15,7 +15,7 @@ class CatalogController < ApplicationController
   # These before_filters apply the hydra access controls
   before_filter :enforce_show_permissions, only: :show
   # This applies appropriate access controls to all solr queries
-  CatalogController.search_params_logic += [:add_access_controls_to_solr_params, :add_advanced_parse_q_to_solr]
+  CatalogController.search_params_logic = [:add_advanced_parse_q_to_solr] + search_params_logic + [:add_access_controls_to_solr_params]
```

Comment:
@jcoyne add_advanced_parse_q_to_solr changes the query and my :show_works_or_works_that_conatin_files also changes the query.  Add add_advanced_parse_q_to_solr just overrides any changes I make, while my change keeps the original string.